t/01-html-checks.t

```diff
@@ -0,0 +1,106 @@
+#!perl
+
+use strict;
+use warnings;
+
+use File::Find;
+use Test::More;
+eval { use HTML::Lint; 1 } or plan skip_all => "HTML::Lint not available";
+
+my @on_disk = ();
+my $is_todo = '';
+
+sub collect {
+    return if $File::Find::name !~ m/\.html$/ || $File::Find::name =~ m/\/pod\//;
+
+    my $module = $File::Find::name;
+    push @on_disk, $module
+}
+find(\&collect, 'UI/');
+
+sub content_test {
+    my ($filename) = @_;
+    my $ui_header_used = 0;
+    $ui_header_used = 1 if $filename =~ m/UI\/lib\//;
+
+    my ($fh, @tab_lines, @trailing_space_lines);
+    open $fh, "<$filename";
+    while (<$fh>) {
+        push @tab_lines, ($.) if /\t/;
+        push @trailing_space_lines, ($.) if / $/;
+        $ui_header_used = 1 if /ui-header\.html/;
+    }
+    close $fh;
+
+TODO: {
+    local $TODO = 'LedgerSMB HTML files is still under development';
+    $is_todo = Test::More->builder->todo;
+    ok((! @tab_lines) && (! @trailing_space_lines),
+        "Source critique for '$filename'");
+    fail("Line# with tabs: " . (join ', ', @tab_lines))
+        if @tab_lines;
+    fail("Line# with trailing space(s): " . (join ', ', @trailing_space_lines))
+        if @trailing_space_lines;
+
+    my $lint = HTML::Lint->new;
+    $lint->only_types(); # Get all
+
+    # Add dojo attributes
+
+    for my $dojo ( 'data-dojo-attach-event',
+                   'data-dojo-attach-point',
+                   'data-dojo-id',
+                   'data-dojo-mixins',
+                   'data-dojo-obj',
+                   'data-dojo-props',
+                   'data-dojo-textdir',
+                   'data-dojo-type',
+                   'data-myscope-id',
+                   'data-myscope-props',
+                   'data-myscope-type') {
+        for my $tag ( 'body', 'button', 'center', 'div', 'fieldset', 'form', 'h1', 'h2', 'h3', 'h4',
```

Comment:
@ylavoie from what I know about HTML5 (which isn't all that much), I think the idea is to attach all non-semantic information to ```data-*``` named attributes, so any ```data-``` attribute is valid. Is there a syntax to declare that?